File: Bridge/Fariyon-03/Bridge/Bridge/Shape/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bridge
{
    public class Rectangle : Shape
    {
        private double height;
        private double width;

        public Rectangle(double height, double width)
        {
            if (height < 0 || width < 0)
                throw new ArgumentException("The height and width must be non-negative");
            this.height = height;
            this.width = width;
        }

        public override double CalculateArea()
        {
            return width * height;
        }

        public override double CalculatePerimeter()
        {
            return 2 * (width + height);
        }
    }
}

File: Bridge/Predko/Shape/Rectangle.java
package org.example.Predko.Shape;

import org.example.Predko.Color.Color;

public class Rectangle extends Shape {
    protected double length;
    private double width;

    public Rectangle(double length, double width, Color color) {
        super(color);
        this.length = length;
        this.width = width;
    }

    public double calculateArea() {
        return length * width;
    }

    public double calculatePerimeter() {
        return 2 * (length + width);
    }
}